app/models/task.py

```diff
@@ -1,5 +1,36 @@
-from sqlalchemy.orm import Mapped, mapped_column
+from sqlalchemy.orm import Mapped, mapped_column, relationship
+from sqlalchemy import ForeignKey
+from typing import Optional, TYPE_CHECKING
 from ..db import db
+from datetime import datetime
+if TYPE_CHECKING:
+    from .goal import Goal
 
 class Task(db.Model):
     id: Mapped[int] = mapped_column(primary_key=True, autoincrement=True)
+    title: Mapped[str]
+    description: Mapped[str]
+    completed_at: Mapped[Optional[datetime]] = mapped_column(nullable=True)
+    goal_id: Mapped[Optional[int]] = mapped_column(ForeignKey("goal.id"))
+    goal: Mapped[Optional["Goal"]] = relationship(back_populates="tasks")
+
+    @classmethod
+    def from_dict(cls, task_data):
+        completed_at = task_data.get("completed_at")
+        new_task = cls(title=task_data["title"],
+                       description=task_data["description"]
+        )
+        
+        return new_task
+
+    def to_dict(self):
+        task_as_dict = {}
+        task_as_dict["id"] = self.id
+        task_as_dict["title"] = self.title
+        task_as_dict["description"] = self.description
+        task_as_dict["is_complete"] = self.completed_at is not None
```

Comment:
For readability and less repetition, I'd opt for this: 
```suggestion
        task_as_dict = {
            "id": self.id,
            "title": self.title,
            "description": self.description,
            "is_complete": self.completed_at is not None
        }
```